src/phases/game-over-phase.ts

```diff
@@ -108,7 +114,9 @@ export class GameOverPhase extends BattlePhase {
             this.scene.gameData.gameStats.dailyRunSessionsWon++;
           }
         }
-        this.scene.gameData.saveRunHistory(this.scene, this.scene.gameData.getSessionSaveData(this.scene), this.victory);
+        this.getRunHistoryEntry().then(runHistoryEntry => {
+          this.scene.gameData.saveRunHistory(this.scene, runHistoryEntry, this.victory);
+        });
```

Comment:
Since `getRunHistoryEntry()` is async, we need to be careful so that the rest of the game over process (specifically, `PostGameOverPhase` doesn't delete the session data before `getRunHistoryEntry()` is finished retrieving and processing the session data.

One potential solution would be to keep track of the Promise:
```suggestion
        const runHistoryEntryPromise = this.getRunHistoryEntry().then(runHistoryEntry => {
          this.scene.gameData.saveRunHistory(this.scene, runHistoryEntry, this.victory);
        });
```

Then, replace the `this.end()` later with `runHistoryEntryPromise.then(() => this.end())`.